src/topology/noetherian_space.lean

```diff
@@ -0,0 +1,220 @@
+/-
+Copyright (c) 2022 Andrew Yang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Andrew Yang
+-/
+import order.compactly_generated
+import order.order_iso_nat
+import topology.sets.compacts
+
+/-!
+# Noetherian space
+
+A Noetherian space is a topological space that satisfies any of the following equivalent conditions:
+- `well_founded ((>) : opens α → opens α → Prop)`
+- `well_founded ((<) : closeds α → closeds α → Prop)`
+- `∀ s : set α, is_compact s`
+- `∀ s : opens α, is_compact s`
+
+The first is chosen as the definition, and the equivalence is shown in
+`topological_space.noetherian_space_tfae`.
+
+Many examples of noetherian spaces come from algebraic topology. For example, the underlying space
+of a noetherian scheme (and spectrums of noetherian rings in particular) is noetherian.
```

Comment:
I find that staying in singular, instead of switching to plural and back, reads slightly more natural.
```suggestion
of a noetherian scheme (e.g., the spectrum of a noetherian ring) is noetherian.
```